Add warnings for `sh` for non-POSIX extensions

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related


#### Here's a snippet or screenshot that shows the problem:

This is just a small example, please read **More details** section below.

```sh
#!/bin/sh 
[ -k foo ]
```

#### Here's what shellcheck currently says:
No issues detected!


#### Here's what I wanted or expected to see:
In POSIX sh, -k is undefined.

(Or something similar.)

#### More details

Shellcheck does not emit warnings for the non-POSIX operators `-nt`, `-ef`, `-k`, `-o`, `<`, and `>` when using a `#!/bin/sh` shebang or `--shell sh`.

Shellcheck already emits warnings for non-POSIX syntax ([example 1](https://github.com/koalaman/shellcheck/wiki/SC2039), [example 2](https://github.com/koalaman/shellcheck/wiki/SC3026)). It would be useful if shellcheck could also warn for these non-POSIX operators, when the `sh` shell is specified.

--- a/src/ShellCheck/Checks/ShellSupport.hs
+++ b/src/ShellCheck/Checks/ShellSupport.hs
@@ -220,6 +220,9 @@ prop_checkBashisms125 = verifyNot checkBashisms "#!/bin/busybox sh\ntype -p test
 prop_checkBashisms126 = verifyNot checkBashisms "#!/bin/busybox sh\nread -p foo -r bar"
 prop_checkBashisms127 = verifyNot checkBashisms "#!/bin/busybox sh\necho -ne foo"
 prop_checkBashisms128 = verify checkBashisms "#!/bin/dash\ntype -p test"
+prop_checkBashisms129 = verify checkBashisms "#!/bin/sh\n[ -k /tmp ]"
+prop_checkBashisms130 = verifyNot checkBashisms "#!/bin/dash\ntest -k /tmp"
+prop_checkBashisms131 = verify checkBashisms "#!/bin/sh\n[ -o errexit ]"
 checkBashisms = ForShell [Sh, Dash, BusyboxSh] $ \t -> do
     params <- ask
     kludge params t
@@ -254,6 +257,18 @@ checkBashisms = ForShell [Sh, Dash, BusyboxSh] $ \t -> do
     bashism (T_SimpleCommand id _ [asStr -> Just "test", lhs, asStr -> Just op, rhs])
         | op `elem` [ "<", ">", "\\<", "\\>", "<=", ">=", "\\<=", "\\>="] =
             unless isDash $ warnMsg id 3012 $ "lexicographical " ++ op ++ " is"
+    bashism (TC_Unary id _ op _)
+        | op `elem` [ "-k", "-G", "-O" ] =
+            unless isDash $ warnMsg id 3013 $ op ++ " is"
+    bashism (T_SimpleCommand id _ [asStr -> Just "test", asStr -> Just op, _])
+        | op `elem` [ "-k", "-G", "-O" ] =
+            unless isDash $ warnMsg id 3013 $ op ++ " is"
+    bashism (TC_Unary id _ op _)
+        | op `elem` [ "-N", "-o", "-R" ] =
+            warnMsg id 3013 $ op ++ " is"
+    bashism (T_SimpleCommand id _ [asStr -> Just "test", asStr -> Just op, _])
+        | op `elem` [ "-N", "-o", "-R" ] =
+            warnMsg id 3013 $ op ++ " is"
     bashism (TC_Binary id SingleBracket op _ _)
         | op `elem` [ "-ot", "-nt", "-ef" ] =
             unless isDash $ warnMsg id 3013 $ op ++ " is"

